core/solver/bicg.cpp

```diff
@@ -120,8 +121,9 @@ void Bicg<ValueType>::apply_impl(const LinOp* b, LinOp* x) const
 
 
 template <typename ValueType>
-void Bicg<ValueType>::apply_dense_impl(const matrix::Dense<ValueType>* dense_b,
-                                       matrix::Dense<ValueType>* dense_x) const
+template <typename VectorType>
+void Bicg<ValueType>::apply_dense_impl(const VectorType* dense_b,
+                                       VectorType* dense_x) const
```

Comment:
This is using the non-distributed dispatch, so it will throw if you pass a distributed vector in.
I guess I would have to revisit that if we make the distributed dispatch the default.